omc3/optics_measurements/phase.py

```diff
@@ -178,16 +190,23 @@ def write_special(meas_input, phase_advances, plane_tune, plane):
         elems_to_bpms = -mus1.loc[minmu1] - mus2.loc[minmu2]
         ph_result = ((bpm_phase_advance + elems_to_bpms) * bd)
         model_value = (model_value * bd) % 1
-        model_desc = f"{elem1} to {elem2} MODEL: {model_value:8.4f}     {'':6s} = {_to_deg(model_value):6.2f} deg"
-        result_desc = (f"{elem1} to {elem2} MEAS : {ph_result % 1:8.4f}  +- {bpm_err:6.4f} = "
-                       f"{_to_deg(ph_result):6.2f} +- {bpm_err * 360:3.2f} deg ({bpm_phase_advance:8.4f} + {elems_to_bpms:8.4f} [{minmu1}, {minmu2}])")
-        lines.extend([model_desc, result_desc])
-    with open(join(meas_input.outputdir, f"special_phase_{plane.lower()}.txt"), 'w') as spec_phase:
-        spec_phase.write('Special phase advances\n')
-        for line in lines:
-            spec_phase.write(line + '\n')
-
-
+        special_phase_df=special_phase_df.append(dict(zip(special_phase_columns,[
+                                                            elem1,
+                                                            elem2,
+                                                            ph_result % 1,
+                                                            bpm_err,
+                                                            _to_deg(ph_result),
+                                                            bpm_err * 360,
+                                                            model_value,
+                                                            _to_deg(model_value),
+                                                            minmu1,
+                                                            minmu2,
+                                                            bpm_phase_advance,
+                                                            elems_to_bpms,
+        ])), ignore_index=True)
+
+    tfs.write(join(meas_input.outputdir, f"special_phase_{plane.lower()}.tfs"), special_phase_df)
```

Comment:
"special_phase_" could be a constant, ".tfs" is already a constant in optics_measurements/constants.py